torch/utils/checkpoint.py

```diff
@@ -96,7 +96,14 @@ def backward(ctx, *args):
 
         if isinstance(outputs, torch.Tensor):
             outputs = (outputs,)
-        torch.autograd.backward(outputs, args)
+        outputs_with_grad = []
+        args_with_grad = []
+        # run backward() with only tensor that requires grad
+        for i in range(len(outputs)):
+            if outputs[i].requires_grad:
+                outputs_with_grad.append(outputs[i])
+                args_with_grad.append(args[i])
+        torch.autograd.backward(outputs_with_grad, args_with_grad)
```

Comment:
What if `len(outputs_with_grad) == 0`?

Reply:
conceptually if none of the output requires grad then checkpoint shouldn't be used in the first place. I will raise an exception if all output doesn't require grad.

Reply:
the current behavior is throwing an warning in check_backward_validity(), though in fact backward pass would actually fail without my change. 
should I keep the warning or change that to an exception?